cmd/pinsharesvc/config.go

```diff
@@ -0,0 +1,310 @@
+	dataDir, err := getUserDataDirectory()
+	if err != nil {
+		return nil, fmt.Errorf("failed to determine data directory: %w", err)
+	}
+
+	configPath := filepath.Join(dataDir, "config.json")
+
+	data, err := os.ReadFile(configPath)
+	if err != nil {
+		return nil, fmt.Errorf("failed to read config file: %w", err)
+	}
+
+	config := &ServiceConfig{}
+	if err := json.Unmarshal(data, config); err != nil {
+		return nil, fmt.Errorf("failed to parse config file: %w", err)
+	}
+
+	config.applyDefaults()
+	return config, nil
+}
+
+// getDefaultConfig returns a configuration with default values
+func getDefaultConfig() (*ServiceConfig, error) {
+	programFiles := os.Getenv("PROGRAMFILES")
+	if programFiles == "" {
+		programFiles = `C:\Program Files`
+	}
+
+	installDir := filepath.Join(programFiles, "PinShare")
+
+	// Get user data directory from session marker or LOCALAPPDATA
+	dataDir, err := getUserDataDirectory()
+	if err != nil {
+		return nil, fmt.Errorf("failed to determine data directory: %w", err)
+	}
+
+	config := &ServiceConfig{
+		InstallDirectory: installDir,
+		DataDirectory:    dataDir,
+		IPFSBinary:       filepath.Join(installDir, "ipfs.exe"),
+		PinShareBinary:   filepath.Join(installDir, "pinshare.exe"),
+
+		IPFSAPIPort:     winservice.DefaultIPFSAPIPort,
+		IPFSGatewayPort: winservice.DefaultIPFSGatewayPort,
+		IPFSSwarmPort:   winservice.DefaultIPFSSwarmPort,
+		PinShareAPIPort: winservice.DefaultPinShareAPIPort,
+		PinShareP2PPort: winservice.DefaultPinShareP2PPort,
+		UIPort:          winservice.DefaultUIPort,
+
+		OrgName:   "MyOrganization",
+		GroupName: "MyGroup",
+
+		SkipVirusTotal: false, // Default to enabled; note: without VT_TOKEN, scanning is auto-skipped in service context
+		EnableCache:    true,
+		ArchiveNode:    false,
+
+		EncryptionKey: generateEncryptionKey(),
+
+		LogLevel:    "info",
+		LogFilePath: filepath.Join(dataDir, "logs", "service.log"),
+	}
+
+	return config, nil
+}
+
+// applyDefaults fills in missing configuration values with defaults
+func (c *ServiceConfig) applyDefaults() {
+	if c.IPFSAPIPort == 0 {
+		c.IPFSAPIPort = winservice.DefaultIPFSAPIPort
+	}
+	if c.IPFSGatewayPort == 0 {
+		c.IPFSGatewayPort = winservice.DefaultIPFSGatewayPort
+	}
+	if c.IPFSSwarmPort == 0 {
+		c.IPFSSwarmPort = winservice.DefaultIPFSSwarmPort
+	}
+	if c.PinShareAPIPort == 0 {
+		c.PinShareAPIPort = winservice.DefaultPinShareAPIPort
+	}
+	if c.PinShareP2PPort == 0 {
+		c.PinShareP2PPort = winservice.DefaultPinShareP2PPort
+	}
+	if c.UIPort == 0 {
+		c.UIPort = winservice.DefaultUIPort
+	}
+	if c.LogLevel == "" {
+		c.LogLevel = "info"
+	}
+	if c.OrgName == "" {
+		c.OrgName = "MyOrganization"
+	}
+	if c.GroupName == "" {
+		c.GroupName = "MyGroup"
+	}
+	if c.EncryptionKey == "" {
+		c.EncryptionKey = generateEncryptionKey()
+	}
+
+	// Set default paths if not specified
+	if c.DataDirectory == "" {
+		dataDir, err := getUserDataDirectory()
+		if err != nil {
+			// Fall back to a reasonable default
```

Comment:
This seems redundant with the definition of `getUserDataDirectory`,ets remove anu duplicate code here and perhaps just handle this error instead.